src/JSInterop/Microsoft.JSInterop/src/DotNetDispatcher.cs

```diff
@@ -172,74 +171,63 @@ private static object[] ParseArguments(string methodIdentifier, string argsJson,
                 return Array.Empty<object>();
             }
 
-            // There's no direct way to say we want to deserialize as an array with heterogenous
-            // entry types (e.g., [string, int, bool]), so we need to deserialize in two phases.
-            var jsonDocument = JsonDocument.Parse(argsJson);
-            var shouldDisposeJsonDocument = true;
-            try
+            var utf8JsonBytes = Encoding.UTF8.GetBytes(argsJson).AsSpan();
```

Comment:
You should be able to avoid this allocation by stackalloc/renting and using
https://github.com/dotnet/corefx/blob/7431e4938c461f6e24cf3122c0b9ca4eb27e4dde/src/Common/src/CoreLib/System/Text/Unicode/Utf8.cs#L42

Reply:
This project targets netstandard2.0. I don't that API exists?

Reply:
Or even better
https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Text/UTF8Encoding.cs#L371

Reply:
Is cross-compiling/multi-targeting feasible? If perf/allocation is super critical, maybe use the char* overload for Encoding.UTF8 on older platforms and the span one on newer ones.

Reply:
@pranavkm We can stackalloc on netstandard2.0, it just has to be done inside a method marked as unsafe. I think we'd be OK with the "unsafe" for now (this will move to netstandard2.1 later). However I'm not sure if additional checks are needed to avoid problems if we receive an especially long string. Brennan's other suggestion may be better!